devicetypes/smartthings/dawon-zwave-smart-switch.src/dawon-zwave-wall-smart-switch.groovy

```diff
@@ -0,0 +1,392 @@
+		result << createEvent(map)
+		result
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	log.info "Crc16Encap called: cmd '${cmd}'"
+	def versions = commandClassVersions
+	def version = versions[cmd.commandClass as Integer]
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	}
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.sensormultilevelv5.SensorMultilevelReport cmd) {
+	log.info "SensorMultilevelReport called, ${cmd}"
+	def map = [:]
+	def result = []
+	switch (cmd.sensorType) {
+		case 1:
+			map.name = "temperature"
+			def cmdScale = cmd.scale == 1 ? "F" : "C"
+			map.value = convertTemperatureIfNeeded(cmd.scaledSensorValue, cmdScale, cmd.precision)
+			map.unit = getTemperatureScale()
+			break		
+		case 5:
+			map.name = "humidity"
+			map.value = cmd.scaledSensorValue.toInteger()
+			map.unit = "%"
+			break
+		default:
+			map.descriptionText = cmd.toString()
+	}
+	log.debug "SensorMultilevelReport, ${map}, ${map.name}, ${map.value}, ${map.unit}"
+	result << createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.versionv1.VersionReport cmd) {
+	log.info "VersionReport called, ${cmd}"
+	def fw = "${cmd.applicationVersion}.${cmd.applicationSubVersion}"
+	updateDataValue("fw", fw)
+	def text = "${device.displayName}: firmware version: $fw, Z-Wave version: ${cmd.zWaveProtocolVersion}.${cmd.zWaveProtocolSubVersion}"
+	createEvent(descriptionText: text, isStateChange: false)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd, endpoint = null) {
+	log.info "***** Unhandled Command called, cmd '${cmd}', endpoint '${endpoint}' *****"
+	[descriptionText: "Unhandled $device.displayName: $cmd", isStateChange: true]
+}
+
+
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+	log.info "ping called"
+	refresh()
+}
+
+def refresh(endpoint) {
+	log.info "refresh called: endpint '${endpoint}' "
+	if(endpoint) {
+		delayBetween([
+		secureEncap(zwave.basicV1.basicGet(), endpoint),
+		"delay 500"
+		], 500)
+	} else {
+		def commands = []
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 1)
+		commands << zwave.sensorMultilevelV5.sensorMultilevelGet(sensorType: 5)
+		sendCommands(commands,1000)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd){
+	log.info "ConfigurationReport called: ${cmd}"
+	def result = []
+	def map = [:]
+	switch (cmd.parameterNumber) {
+		case 1:
+			state.tempOffset = cmd.scaledConfigurationValue
+			break
+		case 2:
+			state.humidityOffset = cmd.scaledConfigurationValue
+			break
+	}
+	log.debug "ConfigurationReport event: ${map}"
+	result <<  createEvent(name: "configReport", value: "${state.tempOffset}, ${state.humidityOffset}%")
```

Comment:
Perhaps this should just be a debug log message?

Reply:
This part will be remove.

Reply:
This part is just log. So. I will delete this part.

Reply:
Just make it a log message, then.

Reply:
Ok. I will update this line to log message.